pytket/extensions/qiskit/qiskit_convert.py

```diff
@@ -480,6 +481,77 @@ def _build_circbox(instr: Instruction, circuit: QuantumCircuit) -> CircBox:
     return CircBox(subc)
 
 
+# Used for handling of IfElseOp
+# docs -> https://docs.quantum.ibm.com/api/qiskit/qiskit.circuit.IfElseOp
+# Examples -> https://docs.quantum.ibm.com/guides/classical-feedforward-and-control-flow
+# pytket-qiskit issue -> https://github.com/CQCL/pytket-qiskit/issues/415
+def _pytket_boxes_from_ifelseop(
+    if_else_op: IfElseOp, qregs: list[QuantumRegister], cregs: list[ClassicalRegister]
+) -> tuple[CircBox, Optional[CircBox]]:
+    # Extract the QuantumCircuit implementing true_body
+    if_qc: QuantumCircuit = if_else_op.blocks[0]
+
+    if_builder = CircuitBuilder(qregs, cregs)
+    if_builder.add_qiskit_data(if_qc)
+    if_circuit = if_builder.circuit()
+    if_circuit.name = "If"
+    # Remove blank wires to ensure CircBox is the correct size.
+    if_circuit.remove_blank_wires()
+
+    # The false_body arg is optional
+    if len(if_else_op.blocks) == 2:
+        else_qc: QuantumCircuit = if_else_op.blocks[1]
+        else_builder = CircuitBuilder(qregs, cregs)
+        else_builder.add_qiskit_data(else_qc)
+        else_circuit = else_builder.circuit()
+        else_circuit.name = "Else"
+        else_circuit.remove_blank_wires()
+        return CircBox(if_circuit), CircBox(else_circuit)
+
+    # If no false_body is specified IfElseOp.blocks is of length 1.
+    # In this case we return a CircBox implementing true_body and None.
+    return CircBox(if_circuit), None
+
+
+def _build_if_else_circuit(
+    if_else_op: IfElseOp,
+    qregs: list[QuantumRegister],
+    cregs: list[ClassicalRegister],
+    qubits: list[Qubit],
+    bits: list[Bit],
+) -> Circuit:
+    # Get two CircBox objects which implement the true_body and false_body.
+    if_box, else_box = _pytket_boxes_from_ifelseop(if_else_op, qregs, cregs)
+    # else_box can be None if no false_body is specified.
+
+    circ_builder = CircuitBuilder(qregs, cregs)
+    circ = circ_builder.circuit()
+
+    # Coniditions must be on a single bit (for now)
+    assert len(if_else_op.condition) == 2
+    if not isinstance(if_else_op.condition[0], Clbit):
```

Comment:
As I said in the description I've restricted to the case of single bit conditions for now. However it should be easy enough to generalise for conditons on values of entire registers in a follow up PR.

Reply:
I prefer the former suggestion as the restriction to a single bit should be removable soon